### DISCUSSION TITLE
emqx_web_hook的事件通知同步还是异步

### DISCUSSION THREAD

**Discussion by chenfeg** [emqx/emqx#6228] [Q&A] [answered]
在docker环境中，配置启用了emqx_web_hook插件，并将事件通知的地址指向了一台提供有web服务同时需要提供mqtt服务的项目（springboot+mqtt），但是当项目启动后，在连接到mqtt时，emqx并将通知发送到了该项目的web服务端口，此时web服务端口并未启动起来，因此mqtt连接得不到响应，导致失败

当然在把mqtt服务改为异步处理时，是可以正常连接的，并收到事件转发通知的

同时，请问如何配置才能过滤掉本服务项目连接时的事件通知，只接收设备连接时的事件通知呢

**Comment by zmstone** [comment, marked as answer]
@chenfeg 
Webhook插件目前没有过滤功能。
可以使用规则引擎过滤：

在 etc/plugins/emqx_rule_engine.conf 中
配置  `rule_engine.events.client_connected = on,qos0`

再添加一条规则

```
SELECT * FROM "$events/client_connected" WHERE clientid != '你的sprintboot+mqtt客户端ID'
```
添加相应的动作触发http post。

**Reply by chenfeg** [reply]
好的，谢谢，我试试
